src/main/java/com/google/api/generator/engine/ast/CommentStatement.java

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class CommentStatement implements Statement {
+  public abstract Comment comment();
+
+  public static CommentStatement withComment(Comment comment) {
+    return builder().setComment(comment).build();
+  }
+
+  @Override
+  public void accept(AstNodeVisitor visitor) {
+    visitor.visit(this);
+  }
+
+  public static Builder builder() {
```

Comment:
We can probably make this method and the Builder class "private".

Reply:
For consistency, should we keep this builder public since there might be chance to call `CommentStatement.builder()` as other classes. But also open to change this, thanks.

Reply:
The `builder()` method and `Builder` class are redundant because we only need `withComment`, could we please make them "private"?

Reply:
Done, thanks! `builder` method is made `private`, while `Builder` class cannot be private because
```
 error: Builder has private access in CommentStatement
  static final class Builder extends CommentStatement.Builder {
```
so it's scoped in package level.